docs/DeveloperGuide.md

```diff
@@ -6,24 +6,177 @@
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddCourse_SequenceDiagram](uml/AddCourse_SequenceDiagram.png)
+
+<br>
+
+When `removeCourse()` is called, it attempts to find the course with the given course code and removes the course 
+from the list of courses if it exists.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![RemoveCourse_SequenceDiagram](uml/RemoveCourse_SequenceDiagram.png)
+
+<br>
+
+There are 3 overloaded variations of the `getPlan()` method in the `PlanGetter` class.
+The first is as shown in the class diagram above. 
+The second takes in an additional integer `year`, while the third takes in two additional integers `year` and `term`.
+
+Each of them loops through the timetable, getting the details of the courses in the course list.
+The first variation outputs the details of all the courses in the plan, 
+while the second and the third only pick out the specified year or term.
+
+<br>
+
+### Storage component
+In order to store user data and load different data files for course planning, 
+PlaNUS uses Storage class for this purpose.
+
+Storage has two main file access APIs:
+- `writeToFile(Timetable)`:
+  - Take in a timetable containing courses, then write courses to the user data file at __./data/myTimetable.csv__.
+    - Params: timetable – A table containing all courses of the user.
+- `Timetable loadTimetable(String)`:
+  - Take in a file name, then load the file containing all courses of the major/user to a Timetable object.
+    - Params: timetableName – The name of the file containing all courses of the major/user. e.g. timetableName of "CEG" indicating the recommended timetable of Computer Engineering, while timetableName of "myTimetable" indicating the timetable of the user.
+    - Returns: A timetable object that is loaded from the given file.
+
+If target file is not found, the above two methods will be able to create the missing file.
+
+Storage also contains a useful parser method `parseCourse(String)` that can parse a line of csv file into a Course object.
+
+Below is a sequence diagram that shows how Storage takes part in the programme.
+
+![Storage_SequenceDiagram](uml/Storage_SequenceDiagram-0.png)
+
+### Grade component
+In order to store and check the grades of courses of the user, 4 classes are used mainly.
+- `Grade`: Represents a grade of a course
+- `Course`: Represents a course taken or planned by the user with grades being specified
+- `Timetable`: Stores the list of courses
+- `GradeChecker`: Provides access to the grades of the given period
+
+Below is the class diagram to illustrate this:
+![Grades_ClassDiagram](uml/Grades_ClassDiagram-0.png)
+
+<br>
+
+When `addGrade()` is called, it finds the course according to the course code. Then, the grade will be set to that course.
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddGrade_SequenceDiagram](uml/AddGrade_SequenceDiagram-0.png)
```

Comment:
The Boolean to check may seem too lengthy. Perhaps, something like matchCourseCode would be easier to read. Then the Boolean can be further explained in a separate comment.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/1376c8a4-d36c-4ff6-b720-c48ba2ca64d1)